terraform/modules/networking/subnet.tf

```diff
@@ -0,0 +1,49 @@
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+locals {
+  nat_gateway_cidr_10 = cidrsubnet(aws_vpc.main.cidr_block, 6, 0) # 0.0/22 - public
+  public_cidr_10      = cidrsubnet(aws_vpc.main.cidr_block, 6, 1) # 4.0/22 - for alb listeners
+  firewall_cidr_10    = cidrsubnet(aws_vpc.main.cidr_block, 6, 2) # 8.0/22
+  private_cidr_10     = cidrsubnet(aws_vpc.main.cidr_block, 6, 3) # 12.0/22- ecs and bastion host
```

Comment:
do we want to consider making this one bigger, as it's the most likely to be expanded into in future? Or are we expecting to expand if necessary into new subnets?

Reply:
Given that this is probably a couple of orders of magnitude more than we'll need based on the current architecture, I think we'd only be at risk of needing more if we add a new feature that might require a lot more IPs, in which case I'd favour putting the new infra in it's own subnet. We can document that in the ADR as well though